csv_detective/explore_csv.py

```diff
@@ -129,53 +105,45 @@ def routine(file_path, num_rows=50, user_input_tests='ALL',output_mode='LIMITED'
+    if not all_tests_fields and all_tests_labels:
         return return_dict
 
-    # Initialising dict for tests
-    test_funcs = dict()
-    for test in all_tests:
-        name = test.__name__.split('.')[-1]
-
-        test_funcs[name] = {
-            'func': test._is,
-            'prop': test.PROPORTION
-        }
-
-    return_table = pd.DataFrame(columns=table.columns)
-    for key, value in test_funcs.items():
-        return_table.loc[key] = table.apply(lambda serie: test_col(
-            serie,
-            value['func'],
-            value['prop'],
-            output_mode=output_mode
-        ))
-
-    # Filling the columns attributes of return dictionnary
-    return_dict_cols = dict()
-    
-    if(output_mode == 'LIMITED'):
-        for colnum in range(0,len(return_table.columns)):
-            col=return_table.columns[colnum]
-            possible_values = list(return_table[return_table[col]].index)
-            if possible_values != []:
-                #print('  >>  La colonne', col, 'est peut-être :',)
-                #print(possible_values)
-                return_dict_cols[header[colnum]] = possible_values
-        return_dict['columns'] = return_dict_cols
-        
-    if(output_mode  == 'ALL'):
-        return_dict_cols = return_table.to_dict('index')
-        return_dict_cols_intermediary = {}
-        for key in return_dict_cols:
-            return_dict_cols_intermediary[key] = []
-            for subkey in return_dict_cols[key]:
-                dict_tmp = {}
-                dict_tmp['colonne'] = subkey
-                dict_tmp['score_rb'] = return_dict_cols[key][subkey]
-                return_dict_cols_intermediary[key].append(dict_tmp)
-        return_dict['columns'] = return_dict_cols_intermediary
+    # Perform testing on fields
+    return_table_fields = test_fields(table, all_tests_fields, output_mode)
+    return_dict_cols_fields = prepare_output_dict(return_table_fields, output_mode)
+    return_dict['columns_fields'] = return_dict_cols_fields
+
+    # Perform testing on labels
+    return_table_labels = test_labels(table, all_tests_labels, output_mode)
+    return_dict_cols_labels = prepare_output_dict(return_table_labels, output_mode)
+    return_dict['columns_labels'] = return_dict_cols_labels
+
+    # Perform a summary by multiplying the two results
+    # The fill_value=1 ensures that if there was no corresponding \
+    #   test for labels and fields, then only the test performed is taken into account
+    return_table = return_table_fields.multiply(return_table_labels, fill_value=1)
+    return_dict_cols = prepare_output_dict(return_table, output_mode)
+    return_dict['columns'] = return_dict_cols
 
     return return_dict
+
+
+if __name__ == "__main__":
+    import os  # for this example only
+    import json  # for json dump only
+    from pathlib import Path
+
+    # Replace by your file path
+    file_path = (Path(os.getcwd()).parent) / 'tests' / 'subventions-commune-de-castelmaurou.csv'
```

Comment:
Why don't retrieve a file furnished in argument with sys.argv ?

Reply:
OK i just removed the main